File: lib/src/main/scala/bruto/variability/primitives/IntegerArgumentVariability.scala
package bruto.variability.primitives

import java.lang.reflect.Type
import bruto.variability.{DiscreteSetWalker, ArgumentVariabilityWalker, ArgumentVariability}

class IntegerArgumentVariability extends ArgumentVariability {

  private val variants: Array[Integer] = Array[Integer](null, 0, 1, 2, Integer.MAX_VALUE)

  override def matches(aType: Type) : Boolean = {
    classOf[Integer] == aType
  }

  override def newWalker(aType: Type) : ArgumentVariabilityWalker = {
    new DiscreteSetWalker(variants)
  }

}

File: lib/src/main/java/bruto/variability/primitives/IntegerArgumentVariability.java
package bruto.variability.primitives;

import bruto.variability.ArgumentVariability;
import bruto.variability.ArgumentVariabilityWalker;
import bruto.variability.DiscreteSetWalker;

import java.lang.reflect.Type;

public class IntegerArgumentVariability extends ArgumentVariability {

    private Integer[] variants = new Integer[]{null, 0, 1, 2, Integer.MAX_VALUE};

    @Override
    public boolean matches(Type aType) {
        return Integer.class == aType;
    }

    @Override
    public ArgumentVariabilityWalker newWalker(Type aType) {
        return new DiscreteSetWalker(variants);
    }

}
